src/lib.rs

```diff
@@ -20,11 +19,7 @@ pub fn sign(
     symbol_name: &str,
 ) -> Result<Vec<u8>, WError> {
     let signature_alg = key_pair.sk.to_alg()?;
-    let module: Module = parity_wasm::deserialize_buffer(module_bytes)?;
-    let module =
-        wasm_signature::attach_signature(module, &signature_alg, ad, &key_pair, symbol_name)?;
-    let signed_module_bytes = parity_wasm::serialize(module)?;
-    Ok(signed_module_bytes)
+    wasm_signature::attach_signature(&module_bytes, &signature_alg, ad, &key_pair, symbol_name)
```

Comment:
I moved serialization/deserialization bits to `wasm_signature` as well for the old signatures to keep things consistent. Let me know if you prefer me to revert this part.

Reply:
No problem! Consistency is great.